code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -0,0 +1,522 @@
+	else if (href_list["stacking_limit"])
+		stacking_limit = input(usr,"Change the threat limit at which round-endings rulesets will start to stack.", "Change stacking limit", null) as num
+	
+	AdminPanel() // Refreshes the window
+
+/datum/game_mode/dynamic/set_round_result()
+	for(var/datum/dynamic_ruleset/rule in executed_rules)
+		if(rule.flags == HIGHLANDER_RULESET)
+			return rule.round_result()
+	return ..()
+
+/datum/game_mode/dynamic/proc/show_threatlog(mob/admin)
+	if(!SSticker.HasRoundStarted())
+		alert("The round hasn't started yet!", "Alert")
+		return
+
+	if(!check_rights(R_ADMIN))
+		return
+
+	var/out = "<TITLE>Threat Log</TITLE><B><font size='3'>Threat Log</font></B><br><B>Starting Threat:</B> [starting_threat]<BR>"
+
+	for(var/entry in threat_log)
+		if(istext(entry))
+			out += "[entry]<BR>"
+
+	out += "<B>Remaining threat/threat_level:</B> [threat]/[threat_level]"
+
+	usr << browse(out, "window=threatlog;size=700x500")
+
+/datum/game_mode/dynamic/proc/generate_threat()
+	relative_threat = lorentz_distribution(curve_centre, curve_width)
+	threat_level = lorentz2threat(relative_threat)
+	threat = round(threat, 0.1)
+
+	peaceful_percentage = round(lorentz_cummulative_distribution(relative_threat, curve_centre, curve_width), 0.01)*100
+
+	threat = threat_level
+	starting_threat = threat_level
+
+/datum/game_mode/dynamic/can_start()
+	message_admins("Dynamic mode parameters for the round: centre = [curve_centre], width is [curve_width]. Extended : [forced_extended], no stacking : [no_stacking].")
+
+	generate_threat()
+
+	latejoin_injection_cooldown = rand(330,510)
+	midround_injection_cooldown = rand(600,1050)
+	message_admins("Dynamic Mode initialized with a Threat Level of... [threat_level]!")
+
+	if (GLOB.player_list.len >= high_pop_limit)
+		message_admins("High Population Override is in effect! Threat Level will have more impact on which roles will appear, and player population less.")
+
+	return TRUE
+
+/datum/game_mode/dynamic/pre_setup()
+	for (var/rule in subtypesof(/datum/dynamic_ruleset/roundstart) - /datum/dynamic_ruleset/roundstart/delayed/)
+		roundstart_rules += new rule()
+	for (var/rule in subtypesof(/datum/dynamic_ruleset/latejoin))
+		latejoin_rules += new rule()
+	for (var/rule in subtypesof(/datum/dynamic_ruleset/midround))
+		var/datum/dynamic_ruleset/midround/DR = rule
+		if (initial(DR.weight))
+			midround_rules += new rule()
+	for(var/mob/dead/new_player/player in GLOB.player_list)
+		if(player.ready == PLAYER_READY_TO_PLAY && player.mind)
+			roundstart_pop_ready++
+			candidates.Add(player)
+	if (candidates.len <= 0)
+		return TRUE
+	if (roundstart_rules.len <= 0)
+		return TRUE
+	
+	roundstart()
+
+	var/starting_rulesets = ""
+	for (var/datum/dynamic_ruleset/roundstart/DR in executed_rules)
+		starting_rulesets += "[DR.name], "
+	return TRUE
+
+/datum/game_mode/dynamic/post_setup(report)
+	for(var/datum/dynamic_ruleset/roundstart/rule in executed_rules)
+		if(rule.execute())
+			continue
+		else
```

Comment:
Else what

Reply:
huh, I were removing some of the admin message spam and I apparently didn't notice that the if-else was there because of admin messaging.